chainntnfs/interface_test.go

```diff
@@ -412,13 +412,172 @@ func testMultiClientConfirmationNotification(miner *rpctest.Harness,
 	}
 }
 
+// Tests the case in which a confirmation notification is requested for a
+// transaction that has already been included in a block. In this case,
+// the confirmation notification should be dispatched immediately.
+func testTxConfirmedBeforeNtfnRegistration(miner *rpctest.Harness,
+	notifier chainntnfs.ChainNotifier, t *testing.T) {
+
+	t.Logf("testing transaction confirmed before notification registration")
+
+	// First, let's send some coins to "ourself", obtainig a txid.
+	// We're spending from a coinbase output here, so we use the dedicated
+	// function.
+
+	txid, err := getTestTxId(miner)
+	if err != nil {
+		t.Fatalf("unable to create test tx: %v", err)
+	}
+
+	// Now generate two blocks. The transaction will be included in the
+	// first block. The notifier must check older blocks when the
+	// confirmation event is registered below to ensure that the TXID hasn't
+	// already been included in the chain, otherwise the notification will
+	// never be sent.
+	if _, err := miner.Node.Generate(1); err != nil {
+		t.Fatalf("unable to generate two blocks: %v", err)
+	}
+
+	// Now that we have a txid, register a confirmation notification with
+	// the chainntfn source.
+	numConfs := uint32(1)
+	confIntent, err := notifier.RegisterConfirmationsNtfn(txid, numConfs)
+	if err != nil {
+		t.Fatalf("unable to register ntfn: %v", err)
+	}
+
+	confSent := make(chan int32)
+	go func() {
+		confSent <- <-confIntent.Confirmed
+	}()
+
+	select {
+	case <-confSent:
+		break
+	case <-time.After(2 * time.Second):
+		t.Fatalf("confirmation notification never received")
+	}
+}
+
+// Tests the case in which a spend notification is requested for a spend that
+// has already been included in a block. In this case, the spend notification
+// should be dispatched immediately.
+func testSpendBeforeNtfnRegistration(miner *rpctest.Harness,
+	notifier chainntnfs.ChainNotifier, t *testing.T) {
+
+	t.Logf("testing spend broadcast before notification registration")
+
+	//createTestSpends(miner *rpctest.Harness, t *testing.T)
```

Comment:
Looks like a left over line from an abandoned refactor, can you remove this line? Thanks!